File: 0739-daily-temperatures/0739-daily-temperatures.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        Stack<int[]> st = new Stack<>();
        int[] ans = new int[temperatures.length];
        for(int i=0;i<temperatures.length;i++){
            while(!st.isEmpty() && temperatures[i]>st.peek()[1]){
                int[] temp = st.pop();
                ans[temp[0]] = i-temp[0];
            }
            st.push(new int[]{i,temperatures[i]});
        }
        return ans;
    }
}

File: 0739-daily-temperatures/0739-daily-temperatures.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        stack = [] #(temp,i)
        ans = [0] * len(temperatures)
        for i,t in enumerate(temperatures):
            while stack and t > stack[-1][0]:
                stackV, stackI = stack.pop()
                ans[stackI] = i - stackI
            stack.append((t,i))
        return ans